File: pe-core/src/main/java/org/saar/core/common/r3d/Model3D.java
package org.saar.core.common.r3d;

import org.saar.core.node.Model;
import org.saar.maths.transform.SimpleTransform;

public class Model3D implements Model {

    private final Mesh3D mesh;

    private final SimpleTransform transform = new SimpleTransform();

    public Model3D(Mesh3D mesh) {
        this.mesh = mesh;
    }

    public SimpleTransform getTransform() {
        return this.transform;
    }

    @Override
    public void draw() {
        this.mesh.draw();
    }

    @Override
    public void delete() {
        this.mesh.delete();
    }
}

File: pe-core/src/main/kotlin/org/saar/core/common/r3d/Model3D.kt
package org.saar.core.common.r3d

import org.saar.core.mesh.Mesh
import org.saar.core.mesh.Model
import org.saar.maths.transform.SimpleTransform

class Model3D(override val mesh: Mesh, val transform: SimpleTransform) : Model {
    constructor(mesh: Mesh) : this(mesh, SimpleTransform())

    var specular: Float = 1f
}